app/V1Module/presenters/SubmitPresenter.php

```diff
@@ -322,6 +322,9 @@ private function finishSubmission(AssignmentSolution $solution, bool $isDebug =
     $submission->setResultsUrl($resultsUrl);
     $this->assignmentSubmissions->persist($submission);
 
+    // The solution needs to reload submissions ...
+    $this->solutions->refresh($solution);
```

Comment:
What exactly is not up to date here?

Reply:
The list of submissions.

Reply:
Ok, so why is the new submission present in the response?

Reply:
And again... as stated before... this is not doctrine way